synapse/handlers/auth.py

```diff
@@ -172,10 +172,14 @@ def __init__(self, hs):
         #   better way to break the loop
         account_handler = ModuleApi(hs, self)
 
-        self.password_providers = [
-            module(config=config, account_handler=account_handler)
-            for module, config in hs.config.password_providers
-        ]
+        self.password_providers = []
+        for module, config in hs.config.password_providers:
+            try:
+                self.password_providers.append(
+                    module(config=config, account_handler=account_handler)
+                )
+            except Exception as e:
+                logger.warn("Error while initializing %r: %s", module, e)
```

Comment:
This would then be something like:

```suggestion
                raise ConfigError("Error while initializing %r: %s" % (module, e))
```

Reply:
Will this work though since we're past the configuration stage when this gets instantiated?

Reply:
That is... true. I'm failing to find an example of us failing while starting up outside of database and config sensibly atm :/